### ISSUE TITLE
False negative in `Layout/LineLength` when line contains 'Http' as constant name

### ISSUE THREAD

**Issue by dari-us** [rubocop/rubocop#15525]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False negative

### Cop name

Layout/LineLength

### Steps to reproduce the problem

I have a type check for a setter, which does not trigger Rubocop despite the line being way too long.
When I use a regular `unless`, Rubocop suggests I should convert it to modifier form.
I suspect that the line length is ignored because the line contains the pattern "Http". However, that is part of a constant name, which caught me by surprise. Doesn't it make more sense to only match String literals containing `http` and `https`?

```ruby
def uploaded_file=(val)
  raise ArgumentError, "uploaded file must be of type ActionDispatch::Http::UploadedFile, given #{val.class.name}" unless val.is_a?(ActionDispatch::Http::UploadedFile)

  @uploaded_file = val
end
```

### Expected behavior

I expect only String literals to be checked for http patterns, not constant names.

### Actual behavior

`Layout/LineLength` ignores this line, because it finds a `http` pattern and therefore seems to think the line content is an URI.
I like that Rubocop can ignore long URIs, but this is definitely not an URI. Also, IMHO it should only account for String literals.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.89.0 (using Parser 3.3.12.0, Prism 1.9.0, rubocop-ast 1.50.0, analyzing as Ruby 4.0, running on ruby 4.0.5) [x86_64-linux]
  - rubocop-capybara 2.23.0
  - rubocop-minitest 0.40.0
  - rubocop-performance 1.26.1
  - rubocop-rails 2.36.0
```
